### ISSUE TITLE
DNSSEC validation

### ISSUE THREAD

**Issue by kisst** [CISOfy/lynis#476]
Check that the DNS resolver validates DNSSEC signatures, or not.
Should be fairly easy to test with either host or dig.

**Comment by mboelen** [comment]
Thanks for the great suggestion. Do you want to make a start with such a test?

**Comment by kisst** [comment]
Sure we can use the existing domains for the test like

```shell
GOOD=$(dig +short sigok.verteiltesysteme.net)
BAD=$(dig +short sigfail.verteiltesysteme.net)

if [ -z "$GOOD" ] && [ -z "$BAD" ]; then
  echo Network error false positive
fi

if [ -z "$GOOD" ] && ! [ -z "$BAD" ]; then
  echo Exception please report it
fi

if ! [ -z "$GOOD" ] && [ -z "$BAD" ]; then
  echo DNSSEC validation OK
fi

if ! [ -z "$GOOD" ] && ! [ -z "$BAD" ]; then
  echo No DNSSEC validation
fi
```
or if the project has some domain which can be used maybe even better. I forked the project now, will review the coding guidelines and will send a PR soon after.

**Comment by kisst** [comment]
Just few things to confirm:

1) this new test would go somewhere around [here](https://github.com/CISOfy/lynis/blob/master/include/tests_networking#L118)

2) What is the `test-no` and `weight` for this test ?

3) You are happy with using "external to the project" DNS records for the validation

4) And special lint or formatting rules in this project 

Thx @mboelen

**Comment by mboelen** [comment]
Hi @kisst - Thanks for your patience, while we are resolving issues one by one.

We had to give this one some thought. As most people are not familiar with the mentioned (external) resource, it might give the project an unwanted dependency on an external DNS name. For now, I suggest to create your own custom test and add it the test to your tests_custom file. Give it a custom ID (e.g. CUST-0010).

Will close this issue for now, to keep the list clean. Feel free to reopen if you see a method to test this without relying on an external resource.

**Comment by kisst** [comment]
ps: @mboelen I can't re-open a ticket in this project ...

**Comment by mboelen** [comment]
Alright, then let's set up records related to the project, on a domain of our own. Please send me an email (on my profile).

1) Let's create a separate DNS test group for it, as I think more tests can be grouped there.
2) Give it number DNS-1600 with low weight
3) Covered
4) See https://github.com/CISOfy/lynis/blob/master/CONTRIBUTING.md

**Comment by mboelen** [comment]
Hi @kisst - any progress on your side?

**Comment by kisst** [comment]
Was on holiday, I have the init on my git stash, have to just clean it up and push it, will try to do it over this weekend. In the meantime will drop you an e-mail about the records needs to be created on your domain.

**Comment by kisst** [comment]
Please check https://github.com/CISOfy/lynis/pull/535 and ping me on keybase or in e-mail about the project specific test DNS records.

**Comment by kisst** [comment]
@mboelen have you had a time to look at the PR above ?

**Comment by mboelen** [comment]
Was merged with #535 - it may need additional work, where one can specify which systems to use for the test instead of predefined value. Will close this ticket for now, as part of a cleanup we do to be able to release the major 3.x version.
